### PR TITLE
setup boost fix

### PR DESCRIPTION
**Pull request by kiritigowda**
* BOOST 1.74

### CODE REVIEW

File: MIVisionX-setup.py [hunk 1]
```diff
@@ -195,7 +195,14 @@
         # package dependencies
         os.system('sudo -v')
         os.system('sudo '+linuxFlag+' '+linuxSystemInstall+' -y ' +
-                  linuxSystemInstall_check+' install libssl-dev libboost-all-dev libboost-dev libboost-system-dev libboost-filesystem-dev')
+                  #linuxSystemInstall_check+' install libssl-dev libboost-all-dev libboost-dev libboost-system-dev libboost-filesystem-dev')
+                  linuxSystemInstall_check+' install libssl-dev')
+        # Boost V 1.74.0 from source
+            os.system('(cd '+deps_dir+'; curl -O -L https://dl.bintray.com/boostorg/release/1.74.0/source/boost_1_74_0.tar.bz2 )')
```
Comment (kiritigowda) [inline MIVisionX-setup.py:201]: Codacy found an issue: [unexpected indent (F999)](https://app.codacy.com/gh/GPUOpen-ProfessionalCompute-Libraries/MIVisionX/pullRequest?prid=6160538)

**Comment by kiritigowda** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/gh/GPUOpen-ProfessionalCompute-Libraries/MIVisionX/pullRequest?prid=6160538&bid=20124018)

**Comment by LakshmiKumar23** [comment]
@kiritigowda @paveltc I tested this on ubuntu18.04 with rocm3.7 docker. And image_augmentation app still fails to build. Can you check once?

File: MIVisionX-setup.py [hunk 2]
```diff
@@ -195,7 +195,14 @@
...
         os.system('sudo '+linuxFlag+' '+linuxSystemInstall+' -y ' +
-                  linuxSystemInstall_check+' install libssl-dev libboost-all-dev libboost-dev libboost-system-dev libboost-filesystem-dev')
+                  #linuxSystemInstall_check+' install libssl-dev libboost-all-dev libboost-dev libboost-system-dev libboost-filesystem-dev')
+                  linuxSystemInstall_check+' install libssl-dev')
+        # Boost V 1.74.0 from source
+        os.system('(cd '+deps_dir+'; curl -O -L https://dl.bintray.com/boostorg/release/1.74.0/source/boost_1_74_0.tar.bz2 )')
+        os.system('(cd '+deps_dir+'; tar xjvf boost_1_74_0.tar.bz2 )')
+        os.system('(cd '+deps_dir+'/boost_1_74_0/; ./bootstrap.sh --prefix=/usr/lib --with-python=python3 )')
+        os.system('(cd '+deps_dir+'/boost_1_74_0/; ./b2 stage -j16 threading=multi )')
+        os.system('(cd '+deps_dir+'/boost_1_74_0/; sudo ./b2 install threading=multi link=static --with-system --with-filesystem)')
```
Comment (LakshmiKumar23) [inline MIVisionX-setup.py:205]: @kiritigowda You need to do a 'sudo ./b2 install threading=multi link=static --with-system --with-filesystem' without the 'link=static' also

Comment (LakshmiKumar23) [reply MIVisionX-setup.py:205]: This only gives the .a files. We need the .so files as well

Comment (rrawther) [reply MIVisionX-setup.py:205]: I think if you don't specify anything, it installs both static and shared libs

Comment (kiritigowda) [reply MIVisionX-setup.py]: I have resolved the static lib build

**Review by LakshmiKumar23 — CHANGES_REQUESTED** [review]
@kiritigowda Can you change it to this and make sure it gets into the ```if``` statement

```wget https://dl.bintray.com/boostorg/release/1.74.0/source/boost_1_74_0.tar.gz
tar -xvf boost_1_74_0.tar.gz
cd boost_1_74_0
./bootstrap.sh --prefix=/usr/local --with-python=python3 && ./b2 stage -j16 threading=multi link=shared
./b2 install threading=multi link=shared
./b2 install threading=multi 
```

**Comment by kiritigowda** [comment]
> @kiritigowda Can you change it to this and make sure it gets into the `if` statement
> 
> ```
> tar -xvf boost_1_74_0.tar.gz
> cd boost_1_74_0
> ./bootstrap.sh --prefix=/usr/local --with-python=python3 && ./b2 stage -j16 threading=multi link=shared
> ./b2 install threading=multi link=shared
> ./b2 install threading=multi 
> ```

@LakshmiKumar23 does the install need sudo?

**Comment by LakshmiKumar23** [comment]
> > @kiritigowda Can you change it to this and make sure it gets into the `if` statement
> > ```
> > tar -xvf boost_1_74_0.tar.gz
> > cd boost_1_74_0
> > ./bootstrap.sh --prefix=/usr/local --with-python=python3 && ./b2 stage -j16 threading=multi link=shared
> > sudo ./b2 install threading=multi link=shared
> >sudo  ./b2 install threading=multi 
> > ```
> 
> @LakshmiKumar23 does the install need sudo?

@kiritigowda Yes, for non-root. Updated above

**Comment by kiritigowda** [comment]
@LakshmiKumar23 & @rrawther - using Boost Version required by MIOpen

**Comment by kiritigowda** [comment]
This commit fixes - RPP Filesystem error, missing python and boost collision - @rrawther @LakshmiKumar23 

```
~/MIVisionX/apps/image_augmentation/build# make
Scanning dependencies of target image_augmentation
[ 50%] Building CXX object CMakeFiles/image_augmentation.dir/image_augmentation.cpp.o
[100%] Linking CXX executable image_augmentation
[100%] Built target image_augmentation
```

**Pull request merged**